velox/functions/sparksql/tests/SplitFunctionsTest.cpp

```diff
@@ -105,5 +106,81 @@ TEST_F(SplitTest, longStrings) {
       {{{"abcdefghijklkmnopqrstuvwxyz"}}});
 }
 
+class StrToMapTest : public SparkFunctionBaseTest {
+ protected:
+  void testStrToMap(
+      const std::vector<StringView>& parameters,
+      const std::vector<std::pair<StringView, std::optional<StringView>>>&
+          expect) {
+    std::string expr;
+    switch (parameters.size()) {
+      case 1:
+        expr = "str_to_map(c0)";
+        break;
+      case 2:
+        expr = "str_to_map(c0, c1)";
+        break;
+      case 3:
+        expr = "str_to_map(c0, c1, c2)";
+        break;
+      default:
+        VELOX_FAIL("Unsupported arguments size");
+    }
+    std::vector<VectorPtr> row;
+    for (int i = 0; i < parameters.size(); ++i) {
+      if (i == 0) {
+        auto col = makeFlatVector<StringView>({parameters[i]});
+        row.push_back(col);
+      } else {
+        auto col = makeConstant<StringView>(parameters[i], 1);
+        row.push_back(col);
+      }
+    }
+    auto result = evaluate<MapVector>(expr, makeRowVector(row));
+    auto expected = makeMapVector<StringView, StringView>({expect});
+    ::facebook::velox::test::assertEqualVectors(result, expected);
+  }
+};
+
+TEST_F(StrToMapTest, Basics) {
+  // clang-format off
```

Comment:
Why do we need this? Can it be clang-formatted?

Reply:
I'm sorry, but the text has become unreadable after it was automatically formatted.